### PR TITLE
Added: Extract internal forces from ReactionsOnly integral

### PR DESCRIPTION
**Pull request by kmokstad**
For calculation of interface forces between patches, instead of using the surface integral which is not that accurate

### CODE REVIEW

File: src/ASM/ReactionsOnly.h [hunk 1]
```diff
@@ -36,7 +36,9 @@ class ReactionsOnly : public GlobalIntegral
   //! \param[in] rf The reaction force vector to be assembled
   //! \param[in] sam Data for FE assembly management
   //! \param[in] adm Parallell processing administrator
-  ReactionsOnly(Vector& rf, const SAM* sam, const ProcessAdm& adm);
+  //! \param[in] sf Internal force vector to be assembled
+  ReactionsOnly(Vector& rf, const SAM* sam, const ProcessAdm& adm,
+                Vector* sf = NULL);
```
Comment (akva2) [inline src/ASM/ReactionsOnly.h:41]: nullptr

Comment (kmokstad) [reply src/ASM/ReactionsOnly.h]: Fixed. Also a similar case in SIMbase.

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=138 ifem-poisson=50 please

File: src/ASM/ASMs3D.C [hunk 2]
```diff
@@ -1720,45 +1663,118 @@ void ASMs3D::getBoundaryNodes (int lIndex, IntVec& nodes,
...
+  int node = this->findStartNode(n1,n2,n3,basis);
+  if (node < 1)
+    return;
+  else if (local)
+    node = 1;
+
+  if (swapW && lEdge <= 8) // Account for swapped parameter direction
+    lEdge += (lEdge-1)%4 < 2 ? 2 : -2;
+
+  std::pair<int,int> range;
```
Comment (akva2) [inline src/ASM/ASMs3D.C:1739]: std::array<int,2> seems more natural to me.

**Review by akva2 — APPROVED** [review]

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-elasticity=138 ifem-poisson=50 please

**Pull request merged**